From bc0b00a06a47ec3277c87a358f072ae9a418b32e Mon Sep 17 00:00:00 2001
From: Nell <xianbin.zhu@spacemit.com>
Date: Thu, 21 Nov 2024 08:43:40 +0800
Subject: [PATCH 1287/1448] k1:dts:hdmi:create a dependency relationship
 between HDMI & HDMI AUDIO using the '-supply' prefix

This method can ensure that audio is initialized after HDMI, the
original method would cause HDMI's power to be turned off during the
system startup process, thereby affecting the debugging of display
functions

Change-Id: I6393f1e160fa90061c5bea39a11d50dc5ffb2a00
---
 arch/riscv/boot/dts/spacemit/k1-x-hdmi.dtsi | 32 ++++++++++-----------
 1 file changed, 15 insertions(+), 17 deletions(-)

diff --git a/arch/riscv/boot/dts/spacemit/k1-x-hdmi.dtsi b/arch/riscv/boot/dts/spacemit/k1-x-hdmi.dtsi
index ec2c6c19734f..e05ce95df1eb 100644
--- a/arch/riscv/boot/dts/spacemit/k1-x-hdmi.dtsi
+++ b/arch/riscv/boot/dts/spacemit/k1-x-hdmi.dtsi
@@ -37,11 +37,22 @@ dpu_offline0_hdmi_out: endpoint@1 {
 		};
 	};
 
+	hdmiaudio: spacemit_snd_sspa@C0883900 {
+		compatible = "spacemit,spacemit-snd-sspa";
+		reg = <0 0xc0883900 0 0x300>,
+			<0 0xc0882000 0 0x50>;
+		clocks = <&ccu CLK_RCPU_HDMIAUDIO>;
+		resets = <&reset RESET_RCPU_HDMIAUDIO>;
+		assigned-clocks = <&ccu CLK_RCPU_HDMIAUDIO>;
+		assigned-clock-rates = <48000>;
+		power-domains = <&power K1X_PMU_DUMMY_PWR_DOMAIN>;
+		function-supply = <&hdmi>;
+		#sound-dai-cells = <0>;
+		status = "disabled";
+	};
+
 	hdmi: hdmi@C0400500 {
-		compatible = "spacemit,hdmi", "simple-bus";
-		#address-cells = <2>;
-		#size-cells = <2>;
-		ranges;
+		compatible = "spacemit,hdmi";
 		reg = <0 0xC0400500 0 0x200>;
 		interrupt-parent = <&intc>;
 		interrupts = <136>;
@@ -61,18 +72,5 @@ hdmi_in: endpoint@0 {
 				remote-endpoint = <&dpu_online2_hdmi_out>;
 			};
 		};
-
-		hdmiaudio: spacemit_snd_sspa@C0883900 {
-			compatible = "spacemit,spacemit-snd-sspa";
-			reg = <0 0xc0883900 0 0x300>,
-				<0 0xc0882000 0 0x50>;
-			clocks = <&ccu CLK_RCPU_HDMIAUDIO>;
-			resets = <&reset RESET_RCPU_HDMIAUDIO>;
-			assigned-clocks = <&ccu CLK_RCPU_HDMIAUDIO>;
-			assigned-clock-rates = <48000>;
-			power-domains = <&power K1X_PMU_DUMMY_PWR_DOMAIN>;
-			#sound-dai-cells = <0>;
-			status = "disabled";
-		};
 	};
 };
-- 
2.47.0

